src/main/java/it/eng/idsa/businesslogic/configuration/SecurityConfig.java

```diff
@@ -62,4 +70,31 @@ public AuthenticationEntryPoint authenticationEntryPoint() {
 		entryPoint.setRealmName("api realm");
 		return entryPoint;
 	}
+
+	@Bean
+	public CorsConfigurationSource corsConfigurationSource() {
+		CorsConfiguration configuration = new CorsConfiguration();
+
+		if (StringUtils.isBlank(allowedOrigins)) {
+			configuration.addAllowedOrigin("*");
+		} else {
+			configuration.setAllowedOrigins(Arrays.asList(allowedOrigins.split(",")));
+		}
+
+		if (StringUtils.isBlank(allowedMethods)) {
+			configuration.addAllowedMethod("*");
+		} else {
+			configuration.setAllowedMethods(Arrays.asList(allowedMethods.split(",")));
+		}
+
+		if (StringUtils.isBlank(allowedHeaders)) {
+			configuration.addAllowedHeader("*");
+		} else {
+			configuration.setAllowedHeaders(Arrays.asList(allowedHeaders.split(",")));
+		}
+
+		UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
+		source.registerCorsConfiguration("/**", configuration);
```

Comment:
Do we need cors for all ecc endpoints or just for API?

Reply:
To get self-description, you're pointing to the root of ECC, so setting only API would cause CORS issues, and that's the reason why CORS is for all endpoints, but also as future proof if there will be any new modifications/features.